test/jdk/java/awt/Focus/CrossFocusRequestTest/CrossFocusRequestTest.java

```diff
@@ -0,0 +1,149 @@
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.AWTException;
+import java.awt.Button;
+import java.awt.Dimension;
+import java.awt.EventQueue;
+import java.awt.FlowLayout;
+import java.awt.Frame;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.TextField;
+import java.awt.event.FocusEvent;
+import java.awt.event.FocusListener;
+import java.awt.event.InputEvent;
+import java.lang.reflect.InvocationTargetException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/*
+ * @test
+ * @key headful
+ * @bug 8307325
+ * @summary Verify the focus owner when non focused windows requesting focus.
+ * @run main CrossFocusRequestTest
+ */
+public class CrossFocusRequestTest {
+
+    private static Frame frame1, frame2;
+    private volatile static Button button;
+    private volatile static TextField textField;
+    private volatile static int waitTimeout = 1000;
+    private static final CountDownLatch clickLatch = new CountDownLatch(1);
+    private static final CountDownLatch focusLatch = new CountDownLatch(1);
+    private volatile static Point compAt;
+    private volatile static Dimension compSize;
+
+    public static void main(String[] args)
+        throws InvocationTargetException, InterruptedException, AWTException {
+
+        try {
+            EventQueue.invokeAndWait(CrossFocusRequestTest::initializeGUI);
+
+            Robot robot = new Robot();
+            robot.setAutoDelay(100);
+            robot.setAutoWaitForIdle(true);
+
+            EventQueue.invokeAndWait(() -> {
+                compAt = button.getLocationOnScreen();
+                compSize = button.getSize();
+            });
+
+            robot.mouseMove(compAt.x + compSize.width / 2,
+                compAt.y + compSize.height / 2);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+
+            robot.waitForIdle();
+            if (!clickLatch.await(waitTimeout, TimeUnit.MILLISECONDS)) {
+                throw new RuntimeException(
+                    "FAIL: Button did not trigger actionPerformed when clicked."
+                        + " Test cannot proceed!");
+            }
+
+            robot.waitForIdle();
```

Comment:
Did you mean to put delays here instead of waitForIdles? I believe you already have setAutoWaitForIdle to handle this. Might be better to set a delay(1000) instead for stability.